components/layout/SideNav.tsx

```diff
@@ -239,114 +295,68 @@ function DocLink(props: DocLinkProps) {
   );
 }
 
-const SectionLinksWrapper = styled.div`
-  padding: 0 20px;
-`;
-
-function SectionLink(props: SideNavLinkProps) {
-  return (
-    <Link href={props.href} passHref>
-      <a
-        css={css`
-          display: flex;
-          height: 30px;
-          align-items: center;
-          justify-content: flex-start;
-          font-family: Poppins;
-          font-size: 14px;
-          color: ${GREY_6};
-          text-decoration: none;
-
-          > div {
-            width: 17px;
-            text-align: center;
-            margin-right: 7px;
-          }
-
-          &:hover {
-            color: ${ORANGE_PRIMARY};
-            text-decoration: underline;
-          }
-        `}
-      >
-        <div>
-          <StyledIcon src={props.iconSrc} alt={props.text} />
-        </div>
-
-        {props.text}
-      </a>
-    </Link>
-  );
-}
-
-export default function SideNav(props: SideNavProps) {
+const SideNav = ({ sectionLinks, pages }: SideNavProps) => {
   const { pathname } = useRouter();
-
-  const [state, dispatch] = useReducer(reducer, getInitialState(pathname));
-
-  const selectedSection = props.links.find(isSelectedSection(pathname));
-
-  const selectedSectionHref = selectedSection?.href;
-  useEffect(() => {
-    if (selectedSectionHref) {
-      dispatch(setMostRecentSectionHref(selectedSectionHref));
-    }
-  }, [selectedSectionHref]);
-
-  const mostRecentSection = props.links.find(
-    hrefIs(state.mostRecentSectionHref)
+  const [activeSection, setActiveSection] = useState(
+    findSelectedSection(sectionLinks, pathname)
   );
+  useEffect(() => {
+    setActiveSection(findSelectedSection(sectionLinks, pathname));
+  }, [pathname]);
```

Comment:
With expanded state handled in the children, using a reducer seemed excessive, so I refactored to use the useState hook instead